pacai/core/distanceCalculator.py

```diff
@@ -54,103 +59,70 @@ def getDistanceOnGrid(self, pos1, pos2):
 
         raise Exception("Position not in grid: " + str(key))
 
-    def isReadyForMazeDistance(self):
-        return (self._distances is not None)
-
-def isInt(pos):
-    x, y = pos
-    return x == int(x) and y == int(y)
-
-def getGrids2D(pos):
-    grids = []
-    for x, xDistance in getGrids1D(pos[0]):
-        for y, yDistance in getGrids1D(pos[1]):
-            grids.append(((x, y), xDistance + yDistance))
-    return grids
-
-def getGrids1D(x):
-    intX = int(x)
-    if x == int(x):
-        return [(x, 0)]
-    return [(intX, x - intX), (intX + 1, intX + 1 - x)]
-
-##########################################
-# MACHINERY FOR COMPUTING MAZE DISTANCES #
-##########################################
-
-distanceMap = {}
-
-class DistanceCalculator:
-    def __init__(self, layout, distancer):
-        self.layout = layout
-        self.distancer = distancer
-        self.cache = {}
-
-    def run(self):
-        if self.layout.walls not in self.cache:
-            self.cache[self.layout.walls] = computeDistances(self.layout)
-
-        self.distancer._distances = self.cache[self.layout.walls]
-
-def computeDistances(layout):
-    """
-    Runs UCS to all other positions from each position.
-    """
-
-    distances = {}
-    allNodes = layout.walls.asList(False)
-
-    for source in allNodes:
-        dist = {}
-        closed = {}
-
-        for node in allNodes:
-            dist[node] = sys.maxsize
-
-        queue = priorityQueue.PriorityQueue()
-        queue.push(source, 0)
-        dist[source] = 0
-
-        while not queue.isEmpty():
-            node = queue.pop()
-            if node in closed:
-                continue
-
-            closed[node] = True
-            nodeDist = dist[node]
-            adjacent = []
-            x, y = node
-
-            if not layout.isWall((x, y + 1)):
-                adjacent.append((x, y + 1))
+    def getGrids2D(self, pos):
+        grids = grid1Dx = grid1Dy = []
+        intX = int(pos[0])
+        intY = int(pos[1])
+        if pos[0] == intX:
+            grid1Dx = [(pos[0], 0)]
+        grid1Dx = [(intX, pos[0] - intX), (intX + 1, intX + 1 - pos[0])]
+        if pos[1] == intY:
+            grid1Dy = [(pos[1], 0)]
+        grid1Dy = [(intY, pos[1] - intY), (intY + 1, intY + 1 - pos[1])]
+        for x, xDistance in grid1Dx:
+            for y, yDistance in grid1Dy:
+                grids.append(((x, y), xDistance + yDistance))
+        return grids
+
```

Comment:
This implementation of ucs is not great. Are we planning on rewriting this, or just switching over to using DP? If so, this comment can be ignored, however, if we do plan on keeping it for a second it needs to be rewritten.